metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/persistence/PersistenceApplicationRunner.java

```diff
@@ -33,48 +34,50 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-public class AnalysisPersistenceApplicationRunner implements ApplicationRunner {
+public class PersistenceApplicationRunner implements ApplicationRunner {
 
-    private static final String ANALYSIS_ACCESSION_FILE_PATH = "analysisAccession.file.path";
+    private static final String ACCESSION_FILE_PATH = "accessions.file.path";
 
-    private static final Logger ANALYSIS_PERSISTENCE_APPLICATION_LOGGER =
-            Logger.getLogger(AnalysisPersistenceApplicationRunner.class.getName());
+    private static final Logger PERSISTENCE_APPLICATION_LOGGER =
+            Logger.getLogger(PersistenceApplicationRunner.class.getName());
 
-    private FileExtractorFromAnalysis fileExtractorFromAnalysis;
+    private StudyObject studyObject;
 
-    public AnalysisPersistenceApplicationRunner() {
-    }
+    private ObjectsImporter objectsImporter;
+
+    private StudyRepository studyRepository;
 
-    public AnalysisPersistenceApplicationRunner(FileExtractorFromAnalysis fileExtractorFromAnalysis) {
-        this.fileExtractorFromAnalysis = fileExtractorFromAnalysis;
+    public PersistenceApplicationRunner(StudyObject studyObject, ObjectsImporter objectsImporter, StudyRepository studyRepository) {
+        this.studyObject = studyObject;
+        this.objectsImporter = objectsImporter;
+        this.studyRepository = studyRepository;
     }
 
     @Override
-    public void run(ApplicationArguments arguments) {
-        Set<String> analysisAccessions = readAccessionsFromFile(arguments);
-        for (String analysisAccession : analysisAccessions) {
-            //TODO Analysis and dependents conversion to metadata model and storing in db
-            List<File> files = fileExtractorFromAnalysis.extractFilesFromAnalysis(analysisAccession);
-        }
-
+    public void run(ApplicationArguments arguments) throws Exception {
+        Set<String> accessions = readAccessionsFromFile(arguments);
+        studyObject.setAccessions(accessions);
+        studyObject = studyObject.accept(objectsImporter);
+        //TODO save study Object importing dependent objects
+        studyRepository.save(studyObject.getStudies());
```

Comment:
I think we concluded that we wouldn't load all the objects in memory at once, but serialize them incrementally? Or are we going for this solution for now, then optimize it if we find it is a bottleneck?

Reply:
I guess for now I have put this way as there are no dependencies saved, I would change it once getting all the dependencies of the study.